### DISCUSSION TITLE
Is ATF deprecated?

### DISCUSSION THREAD

**Discussion by ghost** [airsdk/Adobe-Runtime-Support#1960] [General]
Hi community!
Today my TexturePacker app showed me, that new version of TexturePacker is available (6.0.2), and in release notes it describe, that they remove support for "deprecated and outdated formats ATF and PVR(v2)". So I wondering, still Harman supporting ATF (and atf-viewer), or is ATF really abandoned? Thank you.

**Comment by ajwfrost** [comment]
Well the ATF texture is still supported by AIR and the ATF tools that are part of the SDK .. so it's not really the case to say that it's deprecated or out-dated.

Possibly they're just assuming that because ATF is associated with the Flash Player, which is in a general "end of life" state, it's therefore no longer needed. Feel free to feed back that ATF is still a format that they should support!

thanks

**Comment by B-Interactive** [comment]
When I reached out to CodeAndWeb (the folk who make TexturePacker) today, about the removal of ATF support, I received a response saying that in the almost 1 year since support had been removed, I was only the 2nd person to have "complained" about it.

He acknowledged AIR was still available, so it doesn't seem deprecation is the issue, rather the size of the community demand.

> The problem with the ATF format ist that the codec is based heavily on floating point operations which deliver different outputs on our test machines (several Intel and AMD processors, M1). These introduce a very big overhead for testing and releasing the software. We've removed it after checking our usage data where we found out that less than 0.1% (!) of our users were using ATF.

If this is an issue that is bothering developers, it would seem CodeAndWeb have not received sufficient feedback or usage data to warrant change.

> The easy solution is to install TexturePacker 6.0.1 - it still comes with ATF file format.

Well that's actually not a bad solution.  They still provide the old versions [here](https://www.codeandweb.com/texturepacker/download).

**Reply by Adolio** [reply]
If you're the 2nd person to have "complained" about it, then I'm maybe the 1st one ;)

I'm stuck at using TexturePacker 6.0.1 for a while now...

@ajwfrost Is there any plan to add support for alternative textures container formats like KTX2 or ASTC?

**Reply by ajwfrost** [reply]
The intention of ATF was to have a single package that could be then used for whatever platform the Flash Player happened to be running on, using whatever GPU/API set was in place at the time. I think in practice - and particularly in AIR apps - this may not have been so much of a key use case, as folk would just package the formats that match what they knew would be required for the platforms they're targeting.

So it would be, I think, relatively straightforward to expand the APIs so that we supported other formats. Not sure about KTX2 and the idea of processing things on the fly .. although I guess that should actually be reasonably straightforward with the tools/source they offer .. but for ASTC that doesn't look too tricky as long as we/you know it's supported by the platform.

So maybe:
* Update [`Context3DTextureFormat`](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/Context3DTextureFormat.html) and the [`createTexture()`](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/Context3D.html#createTexture()) method so that we can generate textures that can be populated from these new formats/containers
* Update [`uploadCompressedTextureFromByteArray()`](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/display3D/textures/Texture.html#uploadCompressedTextureFromByteArray()) so that it takes a bytearray using whatever format is expected based on the `createTexture()` call..

That may be relatively straightforwad, although I suspect there are a lot of complexities in here that aren't yet apparent!

thanks

**Comment by bwhiting** [comment]
I might have miss remembered but am fairly sure texture packer can run a custom command after it runs or is runnable via command line. Either way you can probably invoke atf commands yourself that way getting the best of both worlds.
I used to have a file watcher that triggered texture packer then some other post processing so it could work for you that way perhaps.
